apps/web/src/contexts/globalLayout/GlobalSidebar/SidebarDrawerContext.tsx

```diff
@@ -0,0 +1,83 @@
+import {
+  createContext,
+  ReactElement,
+  ReactNode,
+  useCallback,
+  useContext,
+  useMemo,
+  useState,
+} from 'react';
+
+type ActiveDrawer = {
+  content: ReactElement;
+  headerText?: string;
+  drawerName: string;
+  showDoneFooter?: boolean;
+};
+
+type DrawerState = {
+  activeDrawer: ActiveDrawer | null;
+};
+
+const DrawerStateContext = createContext<DrawerState | undefined>(undefined);
```

Comment:
Should we have a discussion about splitting up all of our contexts into two? I don't think it's worth it and just adds extra complexity. I really don't think we'll take a perf hit if we don't do it.

Reply:
will defer to you/ @Robinnnnn  @jakzaizzat  on this one since it seems like a larger decision!